underscore.js

```diff
@@ -449,8 +449,18 @@
 
   // Return the number of elements in an object.
   _.size = function(obj) {
+    var typeObj = toString.call(obj);
     if (obj == null) return 0;
-    return isArrayLike(obj) ? obj.length : _.keys(obj).length;
+    if (isArrayLike(obj)) {
+      return obj.length;
+    } else {
+      switch (typeObj) {
+        case '[object Set]':
```

Comment:
This won't do because IE11 reports them as `[object Object]`, something that `_.isMap` and `_.isSet` can be patched to avoid (out of scope for this PR).